### ISSUE TITLE
Stack overflow when comparing two nil objects of type [String: Any]? 

### ISSUE THREAD

**Issue by DrBreen** [launchdarkly/ios-client-sdk#186]
**Describe the bug**
Extension that overrides `==` operator for `Optional<[String:Any]>` causes infinite loop with stack overflow when comparing two `nil` optionals of this type.

**To reproduce**
1. Add LaunchDarkly 4.0 to Podfile:
`pod 'LaunchDarkly', '~>4.0.0'`
2. Run `pod install`
3. Add following code anywhere in application
```
let dict1: [String: Any]? = nil
let dict2: [String: Any]? = nil

if dict1 == dict2 {
    print("This will never be printed")
}
```
4. Run the application and make sure the code executes.
5. Application will crash due to stack overflow because of line 79 in `LaunchDarkly/Extensions/Dictionary.swift` - it enters recursive loop in case of `lhs` and `rhs` both being nil.

**Expected behavior**
`==` should return true for comparison of two `nil` `[String:Any]?`objects.

**SDK version**
4.0.0

**Language version, developer tools**
Swift 5

**OS/platform**
iOS 12

**Comment by torchhound** [comment]
I just released iOS client 4.1.2 with your fix for this issue, thanks again for the PR!

**Comment by lietusme** [comment]
This seems to be very similarly broken on 4.2.0, but now comparing [String: Any]? with nil:

`class Crash {
    static func test(parameters: [String: Any]? = nil) {
        if parameters == nil {
            print("Yes it's nil")
        }
    }
}`

Usage:
`Crash.test()`

Comparison overflows within this LD extension:
`extension Optional where Wrapped == [String: Any] {
    public static func == (lhs: [String: Any]?, rhs: [String: Any]?) -> Bool {
        guard let lhs = lhs
        else {
            **return rhs == nil**
        }
        guard let rhs = rhs
        else {
            return false
        }
        return lhs.isEqual(to: rhs)
    }

    public static func != (lhs: [String: Any]?, rhs: [String: Any]?) -> Bool {
        return !(lhs == rhs)
    }
}`
